### PR TITLE
Feat/course screening

### PR DESCRIPTION
**Pull request by Fabian5150**
Closes https://github.com/corona-school/project-user/issues/975

Don't merge before https://github.com/corona-school/backend/pull/1079

### CODE REVIEW

**Comment by Fabian5150** [comment]
I initially merged the new main into this branch, but this made the whole pull request changes absolutely chaotic.
Now it really only shows what I did.
Atm there will be a little merge conflict in NavigatorLazy, but if this pr gets approved I can fix it easily

File: src/lang/de.json [hunk 1]
```diff
@@ -2029,14 +2030,42 @@
...
+            "allowed": "Genehmigt",
+            "denied": "Abgelehnt",
+            "cancelled": "Abgesagt",
+            "published": "veröffentlicht",
+            "not_published": "nicht veröffentlicht",
+            "allow_course": "Kurs freischalten",
+            "deny_course": "Kurs ablehnen",
+            "share_course": "Kurs als Vorlage markieren",
+            "unshare_course": "Markierung als Vorlage entfernen",
+            "are_you_shure_allow": "Bist du dir sicher, dass du diesen Kurs freischalten möchtest?",
```
Comment (Jonasdoubleyou) [inline src/lang/de.json:2055]: (sure)

File: src/pages/pupil/SingleCoursePupil.tsx [hunk 2]
```diff
@@ -86,6 +86,7 @@ query GetSingleSubcoursePupil($subcourseId: Int!) {
             name
             }
             allowContact
+            shared
```
Comment (Jonasdoubleyou) [inline src/pages/pupil/SingleCoursePupil.tsx:89]: SubcoursePupil?

File: src/pages/screening/SingleCourseScreener.tsx [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                    {course?.description}
+                </Text>
+            ),
+        },
+    ];
+
+    const courseStatus = () => {
+        let s = '';
+
+        if (course?.courseState === Course_Coursestate_Enum.Allowed) s += t('screening.courses.allowed');
```
Comment (Jonasdoubleyou) [inline src/pages/screening/SingleCourseScreener.tsx:183]: (switch)

File: src/pages/student/SingleCourseStudent.tsx [hunk 4]
```diff
@@ -127,6 +127,7 @@ query GetBasicSubcourseStudent($subcourseId: Int!) {
             name
             }
             allowContact
+            shared
```
Comment (Jonasdoubleyou) [inline src/pages/student/SingleCourseStudent.tsx:130]: same here?

File: src/types/lernfair/User.ts [hunk 5]
```diff
@@ -12,6 +12,7 @@ export type Role =
     | 'USER'
     | 'SCREENER'
     | 'TRUSTED_SCREENER'
+    | 'COURSE_SCREENER'
```
Comment (Jonasdoubleyou) [inline src/types/lernfair/User.ts:15]: (while at it, maybe also add the other two roles)

File: src/pages/student/SingleCourseStudent.tsx (hunk 4 above)
Comment (Jonasdoubleyou) [reply src/pages/student/SingleCourseStudent.tsx:130]: I'm fine to also query it here and for pupils, but then we should check if it is accessible by all roles

File: src/pages/pupil/SingleCoursePupil.tsx (hunk 2 above)
Comment (Fabian5150) [reply src/pages/pupil/SingleCoursePupil.tsx]: that was a very bad solution to overfetch here... 
I've changed the SubcourseData Component now to have the shared property optional here

File: src/pages/student/SingleCourseStudent.tsx (hunk 4 above)
Comment (Fabian5150) [reply src/pages/student/SingleCourseStudent.tsx]: that was a very bad solution to overfetch here... 
I've changed the SubcourseData Component now to have the shared property optional here

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**